src/components/UnstyledButton/tests/UnstyledButton.test.tsx

```diff
@@ -95,202 +95,232 @@ describe('<Button />', () => {
         onKeyUp: noop,
         onKeyPress: noop,
       };
-      const button = mountWithAppProvider(
+      const button = mountWithApp(
         <UnstyledButton url={mockUrl} {...mockUnpassedProps} />,
       );
 
-      expect(button.find(UnstyledLink).prop('url')).toBe(mockUrl);
-
-      expect(button.find(UnstyledLink).prop('submit')).toBeUndefined();
-      expect(button.find(UnstyledLink).prop('ariaControls')).toBeUndefined();
-      expect(button.find(UnstyledLink).prop('ariaExpanded')).toBeUndefined();
-      expect(button.find(UnstyledLink).prop('onKeyDown')).toBeUndefined();
-      expect(button.find(UnstyledLink).prop('onKeyUp')).toBeUndefined();
-      expect(button.find(UnstyledLink).prop('onKeyPress')).toBeUndefined();
+      expect(button).toContainReactComponent(UnstyledLink, {
+        url: mockUrl,
+        submit: undefined,
+        ariaControls: undefined,
+        ariaExpanded: undefined,
+        onKeyDown: undefined,
+        onKeyUp: undefined,
+        onKeyPress: undefined,
+      });
     });
   });
 
   describe('external', () => {
     const mockUrl = 'https://google.com';
 
     it('gets passed into the UnstyledLink', () => {
-      const button = mountWithAppProvider(
-        <UnstyledButton url={mockUrl} external />,
-      );
-      expect(button.find(UnstyledLink).prop('external')).toBeTruthy();
+      const button = mountWithApp(<UnstyledButton url={mockUrl} external />);
+      expect(button).toContainReactComponent(UnstyledLink, {
+        external: true,
+      });
     });
 
     it('is false when not set', () => {
-      const button = mountWithAppProvider(
-        <UnstyledButton url="https://google.com" />,
-      );
-      expect(button.find(UnstyledLink).prop('external')).toBeFalsy();
+      const button = mountWithApp(<UnstyledButton url="https://google.com" />);
+      expect(button).toContainReactComponent(UnstyledLink, {
+        external: undefined,
+      });
     });
 
     it('is not passed when `url` is missing', () => {
-      const button = mountWithAppProvider(<UnstyledButton external />);
-      expect(button.find('button').prop('external')).toBeUndefined();
+      const button = mountWithApp(<UnstyledButton external />);
+      expect(button).toContainReactComponent('button');
```

Comment:
This should test for the absence of `external`

Reply:
I updated the PR with the idea i had in mind, let me know what do you think @BPScott